src/context.rs

```diff
@@ -2396,19 +2404,21 @@ impl ContextWriter {
 
       if row_offset.abs() <= max_row_offs.abs() && row_offset.abs() > processed_rows {
         let found_match = self.scan_row_mbmi(bo, row_offset, max_row_offs, &mut processed_rows, ref_frame, mv_stack,
-                                             &mut far_newmv_count);
+                                             &mut far_newmv_count, bsize);
         row_match |= found_match;
       }
 
       if col_offset.abs() <= max_col_offs.abs() && col_offset.abs() > processed_cols {
         let found_match = self.scan_col_mbmi(bo, col_offset, max_col_offs, &mut processed_cols, ref_frame, mv_stack,
-                                             &mut far_newmv_count);
+                                             &mut far_newmv_count, bsize);
         col_match |= found_match;
       }
     }
 
     let total_match = if row_match { 1 } else { 0 } + if col_match { 1 } else { 0 };
 
+    assert!(total_match >= nearest_match);
```

Comment:
Why? Doesn't seem like it'll be possible unless someone unintentionally breaks the code (at which point a desync should happen).

Reply:
Yes, this won't ever happen. The only reason it's there is to facilitate understanding of the lines below. In particular, the fact that total_match is equal to at least 1 when nearest_match is 1 may not be obvious. Alternatively this could me put in a comment